android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PwoDiscoveryService.java

```diff
@@ -227,7 +291,17 @@ public void onPwoDiscovered(PwoMetadata pwoMetadata) {
     PwoMetadata storedPwoMetadata = mUrlToPwoMetadata.get(pwoMetadata.url);
     if (storedPwoMetadata == null) {
```

Comment:
If storedPwoMetadata is null, then pwoResponseCallback.onPwoDiscovered(storedPwoMetadata) is still called later in this method, receving null as the argument.